Automattic/wp-calypso
# State: Store new password when creating a new application password

This PR is part of #22912 and introduces some changes to allow storing the new password after being created. This is necessary, because we're showing it to the user for a short time. It should be stored separately though, and not in the general app passwords tree, because it's private, and won't be fetched with the application passwords.

So this PR:

* Introduces action creators for:
  * Application password creation success.
  * New application password clearing (our current UI triggers that).
* Updates data layer to:
  * Call the success action when creating an application password.
  * Introduce a validation schema because we'll be storing some information.
  * Use that schema to validate and parse the response.
* Updates the reducer to:
  * Nest the current reducer under a `items` subtree.
  * Add a simple `newPassword` tree to store the new password.
  * Manage the new `newPassword` tree - store and erase the new password.
* Updates and expands tests for all the above accordingly.

To test:
* Checkout this branch
* Verify all tests pass: `npm run test-client application-passwords`
* Fire a `dispatch( { type: 'APPLICATION_PASSWORD_CREATE', applicationName: 'example' } )` in your console, and after receiving has completed, verify the new app password is there (by firing `state.applicationPasswords.items`). Also verify that the new password has been stored in `state.applicationPasswords.newPassword`.
* Fire a `dispatch( { type: 'APPLICATION_PASSWORD_NEW_CLEAR' } )` and verify that the new password has been cleared from `state.applicationPasswords.newPassword`.
